packages/clarity-js/src/layout/region.ts

```diff
@@ -44,13 +44,19 @@ export function exists(node: Node): boolean {
     return regionMap && regionMap.has(node);
 }
 
-export function track(id: number): void {
-    // Do not send another entry if we are already sending it
+export function track(id: number, event: Event): void {
     let node = dom.getNode(id);
-    if (updates.indexOf(id) <= 0 && regionMap.has(node)) {
-        regions[id] = id in regions ? regions[id] : { id, visible: BooleanFlag.False, region: regionMap.get(node) };
-        data.push(regions[id]);
-    }
+    let d = id in regions ? regions[id] : { id, name: regionMap.get(node), state: RegionState.Rendered };
+    let state = event === Event.Click && (d.state !== RegionState.Clicked && d.state !== RegionState.Input) ? RegionState.Clicked : d.state;
```

Comment:
>d.state !== RegionState.Clicked [](start = 42, length = 31)

do we need this check? we would anyway reassign the same value

Reply:
Good point - removed that particular check. Now: Event.Click && d.state !== Interaction.Input ? Interaction.Clicked : d.state;

---
In reply to: [643631746](https://github.com/microsoft/clarity/pull/137#discussion_r643631746) [](ancestors = 643631746)

Reply:
also simplified the whole logic per our offline conversation.